linker.ts

```diff
@@ -1,72 +1,132 @@
 import assert from 'assert';
 import { keccak256 } from 'js-sha3';
+import { isNil, isObject } from './common/helpers';
 
-function libraryHashPlaceholder (input) {
-  return '$' + keccak256(input).slice(0, 34) + '$';
+/**
+ * Newer versions of the compiler use hashed names instead of the string values.
+ *
+ * This helper makes the conversion between name and hash at the target length
+ * for replacement.
+ *
+ * @param libraryName
+ */
+function libraryHashPlaceholder (libraryName) {
```

Comment:
I think docstrings should ideally start with a short explanation of what the function is doing, even if it's terse and requires more explanation - more in-depth explanations can go after that.

```suggestion
/**
 * Generates a new-style library placeholder from a fully-qualified library name.
 *
 * Newer versions of the compiler use hashed names instead of just truncating the name
 * before putting it in a placeholder.
 *
 * @param libraryName
 */
function libraryHashPlaceholder (fullyQualifiedLibraryName) {
```